### PR TITLE
feat: Suggestion product card component

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to add the `SuggestedProductCard` component, related to the [Suggestion dropdown component](https://vtex-dev.atlassian.net/browse/FSSS-202).

**Note:** The component has mock as the integration will be done later.

## How does it work?

Right now, as we do not have the entire component done yet, one way to test it is to render it locally somewhere in the code.

**_Design:_**

Screen Shot 2022-02-22 at 23 34 11
Screen Shot 2022-02-22 at 23 34 05 Screen Shot 2022-02-22 at 23 33 56

**_Result:_**

Screen Shot 2022-02-22 at 23 34 54
Screen Shot 2022-02-22 at 23 36 41
Screen Shot 2022-02-22 at 23 38 07

## References

[FSSS-195 - Suggested Product Card](https://vtex-dev.atlassian.net/jira/software/projects/FSSS/boards/632?selectedIssue=FSSS-195)

## Checklist

- [x] CHANGELOG entry added

### CODE REVIEW

File: src/components/search/SuggestedProductCard/suggested-product-card.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  [data-suggested-product-card-summary] {
+    flex-direction: column;
+    grid-column: 2 / span 4;
+
+    [data-suggested-product-card-title] {
+      display: -webkit-box;
+      display: -moz-box;
+      display: box;
```
Comment (ArthurTriis1) [inline src/components/search/SuggestedProductCard/suggested-product-card.scss:26]: I think the display box can generate some problems, we did a[ PR to fix a similar problem ](https://github.com/vtex-sites/base.store/pull/354/files)

**Review by renatamottam — APPROVED** [review]
Nice job!

**Review by ArthurTriis1 — APPROVED** [review]

**Review by ArthurTriis1 — APPROVED** [review]

File: src/components/search/SuggestedProductCard/SuggestedProductCard.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            breakpoints={[50, 100, 150]}
+            layout="constrained"
+            backgroundColor="#f0f0f0"
+            options={{
+              fitIn: true,
+            }}
+          />
+        </CardImage>
+        <div data-suggested-product-card-summary>
+          <div data-suggested-product-card-title>
```
Comment (hellofanny) [inline src/components/search/SuggestedProductCard/SuggestedProductCard.tsx:59]: Do we really need this wrapper here?

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        overflow: hidden;
+        -webkit-box-orient: vertical;
+        -moz-box-orient: vertical;
+        -webkit-line-clamp: 2;
+        line-clamp: 2;
+      }
+    }
+  }
+
+  [data-suggested-product-card-prices] {
```
Comment (eduardoformiga) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:43]:
```suggestion
  [data-suggestion-product-card-prices] {
```

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      margin-bottom: var(--space-0);
+      color: inherit;
+      line-height: 1.2;
+      text-decoration: none;
+      text-overflow: -o-ellipsis-lastline;
+      text-overflow: ellipsis;
+      outline: none;
+
+      &[data-suggestion-product-card-title] {
+        display: -webkit-box;
```
Comment (hellofanny) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:32]: I think we can extract this part to a mixin, something like `truncate-title` and simplify this part.
We are using this style (line-clamp limiting 2 lines) also on `product-card.scss` and we will need to add it to `CartItem`

**Review by eduardoformiga — APPROVED** [review]
LGTM! ✨ 👏 
HOggKa2b5R

Just a little typo while renaming the component.

**Review by hellofanny — APPROVED** [review]
🚀 Niceeee!

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss [hunk 5]
```diff
@@ -0,0 +1,37 @@
+@import "src/styles/scaffold";
+
+.suggestion-product-card {
+  padding: var(--space-1) var(--space-3);
+
+  [data-card-content] {
+    display: grid;
+    grid-template-columns: rem(56px) repeat(4, 1fr);
+    align-items: center;
+    column-gap: rem(16px);
```
Comment (hellofanny) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:10]:
```suggestion
    column-gap: var(--grid-gap-0);
```

File: src/styles/utilities.scss [hunk 6]
```diff
@@ -53,3 +53,14 @@ $base: 16px !default;
   }
   &:focus-visible { @include focus-ring; }
 }
+
+@mixin truncate-title {
```
Comment (eduardoformiga) [inline src/styles/utilities.scss:57]: 💯 🆒 !

**Review by eduardoformiga — APPROVED** [review]

**Pull request merged**